include/fiction/algorithms/physical_design/graph_oriented_layout_design.hpp

```diff
@@ -753,17 +816,97 @@ class graph_oriented_layout_design_impl
      */
     uint64_t max_placed_nodes{0ul};
     /**
-     * Current best solution w.r.t. area.
+     * The current best solution with respect to area, initialized to the maximum possible value.
+     * This value will be updated as better solutions are found.
+     */
+    uint64_t best_area_solution = std::numeric_limits<uint64_t>::max();
+    /**
+     * The current best solution with respect to the number of wire segments, initialized to the maximum possible value.
+     * This value will be updated as better solutions are found.
+     */
+    uint64_t best_wire_solution = std::numeric_limits<uint64_t>::max();
+    /**
+     * The current best solution with respect to the number of crossings, initialized to the maximum possible
+     * value. This value will be updated as better solutions are found.
+     */
+    uint64_t best_crossing_solution = std::numeric_limits<uint64_t>::max();
+    /**
+     * The current best solution with respect to the area-crossing product (ACP), initialized to the maximum possible
+     * value. This value will be updated as better solutions are found.
+     */
+    uint64_t best_acp_solution = std::numeric_limits<uint64_t>::max();
+    /**
+     * The current best solution with respect to a custom cost objective, initialized to the maximum possible value.
+     * This value will be updated as better solutions are found.
      */
-    uint64_t best_solution{std::numeric_limits<uint64_t>::max()};
+    uint64_t best_custom_solution = std::numeric_limits<uint64_t>::max();
     /**
      * Current best solution w.r.t. area after relocating POs.
      */
     uint64_t best_optimized_solution{std::numeric_limits<uint64_t>::max()};
     /**
-     * Flag indicating if an initial solution was already found, and that other search space graphs should be pruned.
+     * Flag indicating if an initial solution has been found with the layout area as cost objective.
+     * When set to `true`, subsequent search space graphs with the layout area as cost objective can be pruned.
+     */
+    bool improve_area_solution = false;
```

Comment:
Grammatically, I would say "Flag indicating that an ..." for each of these attributes' docstrings.